### PR TITLE
#145 Add warnings and infos to general Tab

### PR DESCRIPTION
**Pull request by micmro**

### CODE REVIEW

File: src/ts/helpers/heuristics.ts [hunk 1]
```diff
@@ -1,7 +1,4 @@
 import {Entry} from "../typing/har";
```
Comment (micmro) [inline src/ts/helpers/heuristics.ts:1]: move the HAR specific heuristics parsing to separate file

File: src/ts/helpers/icons.ts [hunk 2]
```diff
@@ -64,32 +64,6 @@ export function err5xx(x: number, y: number, title: string, scale: number = 1):
     8.4202993,1.7306 8.6878537,1.57 9.0000005,1.57 q 0.3121468,0 0.5797012,0.1606 0.2675544,0.1605 0.4191685,0.437
     z" />`); }
```
Comment (micmro) [inline src/ts/helpers/icons.ts:66]: both were the same error icon

File: src/ts/typing/waterfall.ts [hunk 3]
```diff
@@ -23,6 +26,24 @@ export interface WaterfallEntry {
   segments: WaterfallEntryTiming[];
   rawResource: Entry;
   requestType: RequestType;
+  /** Warnings, Errors and Info indicators  */
+  indicators: WaterfallEntryIndicator[];
```
Comment (micmro) [inline src/ts/typing/waterfall.ts:30]: add indicators to Entries at parse-time

File: src/ts/waterfall/details-overlay/html-details-body.ts [hunk 4]
```diff
@@ -36,6 +36,42 @@ function makeImgTab(accordionHeight: number, entry: WaterfallEntry) {
...
+  let errors = entry.indicators
+    .filter((i) => i.type === "error")
+    .map((i) => [i.title, i.description] as KvTuple);
+  let warnings = entry.indicators
+    .filter((i) => i.type === "warning")
+    .map((i) => [i.title, i.description] as KvTuple);
+  // all others
+  let info = entry.indicators
+    .filter((i) => i.type !== "error" && i.type !== "warning")
+    .map((i) => [i.title, i.description] as KvTuple);
```
Comment (micmro) [inline src/ts/waterfall/details-overlay/html-details-body.ts:58]: separate sections for different indicator types.

File: src/ts/waterfall/svg-chart.ts [hunk 5]
```diff
@@ -136,7 +133,7 @@ export function createWaterfallSvg(data: WaterfallData, options: ChartOptions):
 
   if (options.showIndicatorIcons) {
     const iconsPerBlock = entriesToShow.map((entry: WaterfallEntry) =>
-      indicators.getIndicatorIcons(entry, context.docIsSsl).length);
+      entry.indicators.length > 0 ? 1 : 0);
```
Comment (micmro) [inline src/ts/waterfall/svg-chart.ts:136]: only show one icon, at a time.

File: src/ts/waterfall/row/svg-indicators.ts [hunk 6]
```diff
@@ -30,47 +30,48 @@ export function getMimeTypeIcon(entry: WaterfallEntry): Icon {
...
   }
 
-  if (docIsSsl && !heuristics.isSecure(entry)) {
-    output.push(makeIcon("noTls", "Insecure Connection"));
-  }
+  let combinedTitle = [];
+  let icon = "";
+  const errors = entry.indicators.filter((i) => i.type === "error");
+  const warnings = entry.indicators.filter((i) => i.type === "warning");
+  const info = entry.indicators.filter((i) => i.type !== "error" && i.type !== "warning");
```
Comment (micmro) [inline src/ts/waterfall/row/svg-indicators.ts:58]: Show the icon with the highest priority `error` > `warning` > `info`||customIcon.

The icons title shows all `icon.titles` combined and grouped by type.

**Comment by tobli** [comment]
Sorry for not getting back on this. I think it's awesome!! Found some small regressions, I'll post issues and/or patches as soon as I get some more time.

**Pull request merged**